### PR TITLE
linux: Remove `inode/directory` from supported MIME types

### PR DESCRIPTION
**Pull request by valaphee**
At the moment Zed is handled as default file browser which causes applications like RustRover to open Zed when instead it should open the Gnome files app. And Zed is probably not intended to be an replacement to the Gnome files app for example.

I'm also currently waiting to fix the issue that Zed is not displayed as an "Application" when using "Open with..." on Arch Linux. Which is caused by not setting `APP_ARGS` which should have the value `%F`

Release Notes:

- Linux: Zed will no longer be marked as the default file browser

### CODE REVIEW

**Comment by valaphee** [comment]
@cla-bot check

**Comment by valaphee** [comment]
@cla-bot check

**Comment by ZorinFoss** [comment]
Remove `inode/directory` from MIME types will remove zed from shown in `Open with...` menu in KDE

**Comment by valaphee** [comment]
> Remove `inode/directory` from MIME types will remove zed from shown in `Open with...` menu in KDE

Maybe its the same issue as with Arch Linux, but the actual reason is that desktop entry is wrong:
`Exec=zeditor` should be `Exec=zeditor %F`

**Comment by bilelmoussaoui** [comment]
It should definitely not use `%F` but `%U` instead as the URI handler is what is used for re-opening zed after you successfully logged into your github account for example

**Comment by valaphee** [comment]
What effect should the mime type in the desktop file have, besides the selection for the default handler?

VSCode also had the same problematic https://github.com/microsoft/vscode/issues/114425

**Comment by jansol** [comment]
Looks like this is indeed a [highly contentious topic](https://lists.freedesktop.org/archives/xdg/2021-May/014538.html) and what one could call a "hot mess" in practice.

Kwrite has `InitialPreference=8` to avoid getting set as the automatic default for any MIME type but that is apparently a KDE extension to the desktop entry format, so I doubt it works anywhere else. In that light it makes sense to just remove the mime association entirely. :(

**Comment by bilelmoussaoui** [comment]
but the mime type association has nothing to do with the default apps. The problem with linux is there are billions of different configurations and a ton of people just have mis-configured arch installations or so.

Basically `xdg-mime query default inode/directory` should give you which is the default file manager and `xdg-mime default org.gnome.Nautilus.desktop inode/directory` to set it to Nautilus (or something else)

So basically this PR should be closed and you should fix your system instead.

**Comment by valaphee** [comment]
I mean I agree that the mime type association should not update the default file associations.

But at the moment it also seems to set and overwrite the default, and this behavior will probably not change in the near future.

This is more about a good default configuration and behavior.

**Comment by valaphee** [comment]
Allegedly it also resets after package updates.

Hmm thats funny, the reason why Zed is chosen over Nautilus is because of alphabetical order. The desktop file for `dev.zed.Zed` is before `org.gnome.Nautilus` (for Arch Linux).

**Comment by mikayla-maki** [comment]
I'm going to be conservative and follow VSCode's example here. Let us know by opening a PR if this is the incorrect decision.

**Comment by williamdes** [comment]
This is quite an interesting thread, thank for for removing this second entry execute as.
That said today I started adding zed:// to an url on the web and it does not work. xdg-open does nothing.
I wonder if this is about the `%f`
Maybe I will file an issue, if anyone has pointers on what it could be?
Firefox says it is associated to Zed so that's good.
But no opening of my file.

XFCE, Debian 12.
